shared/src/k8s/mod.rs

```diff
@@ -95,133 +88,78 @@ impl OwnershipInfo {
 #[automock]
 #[async_trait]
 pub trait KubeInterface: Send + Sync {
-    fn get_kube_client(&self) -> APIClient;
+    fn get_kube_client(&self) -> Client;
 
-    async fn find_node(
-        &self,
-        name: &str,
-    ) -> Result<Object<NodeSpec, NodeStatus>, Box<dyn std::error::Error + Send + Sync + 'static>>;
+    async fn find_node(&self, name: &str) -> Result<Node, anyhow::Error>;
 
-    async fn find_pods_with_label(
-        &self,
-        selector: &str,
-    ) -> Result<
-        ObjectList<Object<PodSpec, PodStatus>>,
-        Box<dyn std::error::Error + Send + Sync + 'static>,
-    >;
-    async fn find_pods_with_field(
-        &self,
-        selector: &str,
-    ) -> Result<
-        ObjectList<Object<PodSpec, PodStatus>>,
-        Box<dyn std::error::Error + Send + Sync + 'static>,
-    >;
-    async fn create_pod(
-        &self,
-        pod_to_create: &Pod,
-        namespace: &str,
-    ) -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>>;
-    async fn remove_pod(
-        &self,
-        pod_to_remove: &str,
-        namespace: &str,
-    ) -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>>;
+    async fn find_pods_with_label(&self, selector: &str) -> Result<ObjectList<Pod>, anyhow::Error>;
```

Comment:
soooo much love for this simplification

Reply:
I agree! I think we can put up an issue to change all `Result<(), Box<dyn std::error::Error + Send + Sync + 'static>>` into `anyhow::Result<()>`. I didn't update it everywhere, just where error bubbling up needed to be changed anyways

Reply:
probably a good 'first issue' type of thing too